packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/types/math_expr.rs

```diff
@@ -0,0 +1,273 @@
+use std::collections::HashMap;
+use strum_macros::Display;
+
+use crate::DoenetMLCore;
+
+const BLANK_MATH_OBJECT: &str = "{\"objectType\":\"math-expression\",\"tree\":\"\u{ff3f}\"}";
+/// A mathematical expression represented as a serialized `math-expressions` object.
+#[derive(Debug, Clone, PartialEq, serde::Deserialize)]
+pub struct MathExpr {
+    pub math_object: String,
+}
+
+impl Default for MathExpr {
+    fn default() -> Self {
+        MathExpr {
+            math_object: BLANK_MATH_OBJECT.to_string(),
+        }
+    }
+}
+
+impl std::fmt::Display for MathExpr {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.to_latex(ToLatexParams::default()))
+    }
+}
+
+impl serde::Serialize for MathExpr {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.math_object)
+    }
+}
+
+impl MathExpr {
+    /// Create a `Math` by parsing the string `text` using the `math-expressions` text parser.
+    ///
+    /// Parameters:
+    /// - `text`: the string to be parsed
+    /// - `split_symbols`: if true, split multi-character symbols that do not contain an number
+    ///   into the product of the characters.
+    /// - `function_symbols`: a list of the symbols that will be treated as a function,
+    ///   i.e., one of these symbols followed by arguments in parentheses
+    ///   will be interpreted as apply that function to the arguments (rather than multiplication)
+    pub fn from_text<TXT: AsRef<str>, FnSymbol: AsRef<str>>(
+        text: TXT,
+        split_symbols: bool,
+        function_symbols: &[FnSymbol],
+    ) -> Self {
+        let s = text.as_ref();
+
+        let math_object =
+            match DoenetMLCore::parse_text_into_math(s, split_symbols, function_symbols) {
+                Ok(res) => res,
+                Err(_) => BLANK_MATH_OBJECT.to_owned(),
+            };
+
+        MathExpr { math_object }
+    }
+
+    /// Create a `Math` by parsing the string `latex` using the `math-expressions` latex parser.
+    ///
+    /// Parameters:
+    /// - `latex`: the string to be parsed
+    /// - `split_symbols`: if true, split multi-character symbols that do not contain an number
+    ///   into the product of the characters.
+    /// - `function_symbols`: a list of the symbols that will be treated as a function,
+    ///   i.e., one of these symbols followed by arguments in parentheses
+    ///   will be interpreted as apply that function to the arguments (rather than multiplication)
+    pub fn from_latex<TXT: AsRef<str>, FnSymbol: AsRef<str>>(
+        latex: TXT,
+        split_symbols: bool,
+        function_symbols: &[FnSymbol],
+    ) -> Self {
+        let s = latex.as_ref();
+
+        let math_object =
+            match DoenetMLCore::parse_latex_into_math(s, split_symbols, function_symbols) {
```

Comment:
Is there a reason we're putting these functions on Core?

Reply:
Not sure. Was just blindly following the prototype again! I'll move them off Core.